test/js-api/jsapi.js

```diff
@@ -661,6 +664,112 @@ test(() => {
+    assert_equals(new Global({type:'f32', mutable: false}, 0xf32) instanceof Global, true);
+    assert_equals(new Global({type:'f64', mutable: false}, 0xf64) instanceof Global, true);
+    assert_equals(new Global({type:'i32', mutable: true}, 0x132) instanceof Global, true);
+    assert_equals(new Global({type:'f32', mutable: true}, 0xf32) instanceof Global, true);
+    assert_equals(new Global({type:'f64', mutable: true}, 0xf64) instanceof Global, true);
+}, "'WebAssembly.Global' constructor function");
+
+test(() => {
+    const globalProtoDesc = Object.getOwnPropertyDescriptor(Global, 'prototype');
+    assert_equals(typeof globalProtoDesc.value, "object");
+    assert_equals(globalProtoDesc.writable, false);
+    assert_equals(globalProtoDesc.enumerable, false);
+    assert_equals(globalProtoDesc.configurable, false);
+}, "'WebAssembly.Global.prototype' data property");
+
+test(() => {
+    const globalProtoDesc = Object.getOwnPropertyDescriptor(Global, 'prototype');
+    globalProto = Global.prototype;
+    assert_equals(globalProto, globalProtoDesc.value);
+    assert_equals(String(globalProto), "[object WebAssembly.Global]");
+    assert_equals(Object.getPrototypeOf(globalProto), Object.prototype);
+}, "'WebAssembly.Global.prototype' object");
+
+test(() => {
+    global1 = new Global({type: 'i32'}, 0x132);
+    assert_equals(typeof global1, "object");
+    assert_equals(String(global1), "[object WebAssembly.Global]");
+    assert_equals(Object.getPrototypeOf(global1), globalProto);
+}, "'WebAssembly.Global' instance objects");
+
+test(() => {
+    const lengthDesc = Object.getOwnPropertyDescriptor(globalProto, 'value');
+    assert_equals(typeof lengthDesc.get, "function");
+    assert_equals(typeof lengthDesc.set, "function");
+    assert_equals(lengthDesc.enumerable, false);
+    assert_equals(lengthDesc.configurable, true);
+}, "'WebAssembly.Global.prototype.value' accessor data property");
+
+test(() => {
+    const valueDesc = Object.getOwnPropertyDescriptor(globalProto, 'value');
+    const valueGetter = valueDesc.get;
+    assert_equals(valueGetter.length, 0);
+    assertThrows(() => valueGetter.call(), TypeError);
+    assertThrows(() => valueGetter.call({}), TypeError);
+    assert_equals(typeof valueGetter.call(global1), "number");
+    assert_equals(valueGetter.call(global1), 0x132);
+}, "'WebAssembly.Global.prototype.value' getter");
+
+test(() => {
+    const valueDesc = Object.getOwnPropertyDescriptor(globalProto, 'value');
+    const valueSetter = valueDesc.set;
+    assert_equals(valueSetter.length, 0);
+    assertThrows(() => valueSetter.call(), TypeError);
+    assertThrows(() => valueSetter.call({}), TypeError);
+    assert_equals(valueSetter.call(global1, 1234), undefined);
+    assert_equals(global1.value, 1234);
+}, "'WebAssembly.Global.prototype.value' setter");
+
+test(() => {
+    const valueOfDesc = Object.getOwnPropertyDescriptor(globalProto, 'valueOf');
```

Comment:
Where is it defined that a WbeAssembly.Global should have `valueOf` ?
Right now I see the following https://github.com/WebAssembly/mutable-global/blob/master/proposals/mutable-global/Overview.md#webassemblyglobalprototype--toprimitive--property 
Is it `valueOf` or `[@@toPrimitive]` ?

Reply:
Sorry, that document is old. See https://github.com/WebAssembly/mutable-global/blob/master/document/js-api/index.bs#L726

I'll update.

Reply:
Thanks, it looks like I based my implementation on old stuff... I removed some of my comments related to implementation details of the overview